Expensify/App
# fix: App crash on Android while resuming from background

### Details

This PR fixes an issue where the Android app crashes when the app is resumed from background after permission change.

### Fixed Issues

$ https://github.com/Expensify/App/issues/7008

### Tests


- Lunch app in an Android device.
- Open a conversation with an attachment in it.
- Tap on the attachment to view it in Attachment view
- Tap on download icon to download the attachment
- Kill the app and go to your phone Setting
- Remove all permission for New Expensify
- Disable "Remove permissions and free up space" tag.
- Go back to New Expensify app icon and open it
- App should restart without crash

### QA Steps


- Lunch app in an Android device.
- Open a conversation with an attachment in it.
- Tap on the attachment to view it in Attachment view
- Tap on download icon to download the attachment
- Kill the app and go to your phone Setting
- Remove all permission for New Expensify
- Disable "Remove permissions and free up space" tag.
- Go back to New Expensify app icon and open it
- App should restart without crash

### Tested On


### Screenshots


#### Web

Not applicable

#### Mobile Web

Not applicable

#### Desktop

Not applicable

#### iOS

Not applicable

#### Android


https://user-images.githubusercontent.com/6880914/148737927-f3e8b7ef-cb5f-446e-9e50-53cb4bc19c0c.mp4
